### PR TITLE
Topic recognition - Improved UNet

### PR DESCRIPTION
**Pull request by kmindc**
Name: Mitchell Keefe
Student ID: 45574539

This pull request is for my implementation of Improved UNet on ISIC 2018 Dataset as per Task 1 of the COMP3710 Project Task Sheet.

The average validation dice similarity score after 10 Epochs of 0.84296 was achieved.

The files included are as follow:

README.MD - contains project documentation, i.e. description, results, dependencies, etc.

dataset.py - contains code to load and preprocess the dataset.

modules.py - contains code for the model itself (Improved UNet)

train.py - contains code for training, validating, testing and saving the model

predict.py - contains code to exemplify usage of the trained model

(Edit: could my README.md file please be marked from here :) if possible)

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files, but the predict.py file is empty. (just created an empty file as per the commit history)
- Algorithm almost solves the problem. No testing part. Average Dice Coefficient in validation dataset: 0.84296, no testing score. No data augmentation.
- Correctly using TensorFlow to construct the Improved U-Net models and functions.
- Good coding design.
- Sufficient docstring and comment in the code. References are good.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, no prediction only training -4
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, can't verify if model runs on test set -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no output of results -2
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
